route/route.go

```diff
@@ -69,9 +69,15 @@ func (r *Route) addTarget(service string, targetURL *url.URL, fixedWeight float6
 		t.StripPath = opts["strip"]
 		t.TLSSkipVerify = opts["tlsskipverify"] == "true"
 		t.Host = opts["host"]
-		t.RedirectCode, _ = strconv.Atoi(opts["redirect"])
-		if t.RedirectCode < 300 || t.RedirectCode > 399 {
-			t.RedirectCode = 0
+
+		if opts["redirect"] != "" {
+			t.RedirectCode, err = strconv.Atoi(opts["redirect"])
+			if err != nil {
+				log.Printf("[ERROR] redirect status code should be numeric in 3xx range. Got: %s", opts["redirect"])
```

Comment:
Ideally these kind of errors should prevent the route from being added, but that would take some refactoring.

Reply:
That could be added the route parser but then it would need to treat opts not as an opaque k/v string. I can have a look at this if and when I write a better parser.